scarb/src/ops/resolve.rs

```diff
@@ -178,6 +179,8 @@ pub fn generate_compilation_units(
         .filter(|member| !member.is_cairo_plugin())
         .collect_vec();
     validate_features(&members, enabled_features)?;
+    check_cairo_version_compatibility(&members, ws, ignore_cairo_version)?;
```

Comment:
This cannot be run on members only. Members are all package that are part of the workspace - but **without their dependencies**. This should be run on all dependencies too.

Reply:
Good catch. Used a different approach with `PackageSolutionCollector.warnings`, let me know what you think